connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala

```diff
@@ -43,6 +43,8 @@ import org.apache.spark.util.SerializableConfiguration
 private[sql] class AvroFileFormat extends FileFormat
   with DataSourceRegister with Logging with Serializable {
 
+  AvroFileFormat.registerCustomAvroTypes()
```

Comment:
ditto. Could you provide a test coverage for this line?  The test case on this PR passes without this.